### PR TITLE
order.py updated for payment method and PUT

### LINKED ISSUE #33: Bug: Completing an order doesn't work
When the user clicks **Complete Order** from the cart view, and then selects a payment type, and then clicks on the **Complete Order** button on that dialog, the order does not complete. It is expected for the order to be marked as complete so that it shows up in the **My Orders** view and the cart is emptied.

### PR DESCRIPTION
**Pull request by lsand18**
Description of PR that completes issue here...

## Changes

- updated key names for PUT request
- filter by object, instead of FK in order.py
- remove wrong pipenv from Pipfile
- update try/except block in product model to include ZeroDivisionError exception - this allows products with no ratings to be shown as rating = none. 

## Testing

Description of how to test code...

- [ ]add items to cart
- [] click complete order
- select payment type and click complete order
- page should direct to orders page with new order displayed and cart should be empty

## Related Issues

- Fixes #33,#51, #6

### CODE REVIEW

File: bangazonapi/views/order.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def update(self, request, pk=None):
         """
         customer = Customer.objects.get(user=request.auth.user)
         order = Order.objects.get(pk=pk, customer=customer)
-        order.payment_type = request.data["payment_type"]
+        order.payment_type_id = request.data["payment_type"]
```
Comment (stevebrownlee) [inline bangazonapi/views/order.py]: Do not directly set the foreign key on an object. Follow the company convention of validating the `payment_type_id` sent in the request by using the `.get()` method and catching the `DoesNotExist` method. If it is valid, set the `payment_type` property to the instance.

File: bangazonapi/views/order.py [hunk 2]
```diff
@@ -147,11 +147,11 @@ def list(self, request):
...
         """
         customer = Customer.objects.get(user=request.auth.user)
-        orders = Order.objects.filter(customer=customer).exclude(payment_type_id=None)
+        orders = Order.objects.filter(customer=customer)
 
-        payment = self.request.query_params.get('payment_id', None)
+        payment = self.request.query_params.get('payment_type', None)
         if payment is not None:
-            orders = orders.filter(payment__id=payment)
+            orders = orders.filter(payment_type_id=payment)
```
Comment (stevebrownlee) [inline bangazonapi/views/order.py]: Again, validate the `payment_type_id` using `get()` so you can filter with objects, not integers

**Review by stevebrownlee — CHANGES_REQUESTED** [review]
Do not access the foreign key integers in your logic. Django wants you to work with objects.

File: bangazonapi/views/order.py (hunk 1 above)
Comment (lsand18) [reply bangazonapi/views/order.py]: updated with objects!

File: bangazonapi/views/order.py (hunk 2 above)
Comment (lsand18) [reply bangazonapi/views/order.py]: done

**Review by Justin-Acosta — APPROVED** [review]

**Pull request merged**